src/main/java/io/kestra/plugin/typesense/AbstractTypesenseTask.java

```diff
@@ -0,0 +1,79 @@
+package io.kestra.plugin.typesense;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.media.Schema.RequiredMode;
+import jakarta.validation.constraints.NotNull;
+import java.time.Duration;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+import org.typesense.api.Client;
+import org.typesense.api.Configuration;
+import org.typesense.resources.Node;
+
+@SuperBuilder
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+public class AbstractTypesenseTask extends Task {
+
+    @Schema(
+        title = "The host of the typsense DB",
+        example = "localhost",
+        requiredMode = RequiredMode.REQUIRED
```

Comment:
requiredMode is not needed, we parse the annotation when generating the schema and add required=true if we found the `@NotNull`